pkg/discovery/discovery.go

```diff
@@ -184,9 +189,26 @@ func generateVolumeNodeAffinity(node *v1.Node) (*v1.VolumeNodeAffinity, error) {
 
 // DiscoverLocalVolumes reads the configured discovery paths, and creates PVs for the new volumes
 func (d *Discoverer) DiscoverLocalVolumes() {
+	var discoErrors []error
 	for class, config := range d.DiscoveryMap {
-		d.discoverVolumesAtPath(class, config)
+		err := d.discoverVolumesAtPath(class, config)
+		if err != nil {
+			discoErrors = append(discoErrors, err)
+		}
 	}
+	d.readySync.Lock()
+	d.ready = discoErrors == nil
```

Comment:
Do we need to store an array of errors? Can it just be a bool?

Reply:
Yes in this case it could be 👍